grt: ignore special nets when detecting dirty nets

eder-matheus opened the pull request:
Fixes [OR-1027](https://openroad.atlassian.net/browse/OR-1027).

osamahammad21 reviewed (changes requested):
Thanks @eder-matheus that solves issue [OR-1027](https://openroad.atlassian.net/browse/OR-1027). However, I don't know why you need the extra nets_with_violations_ vector. It only holds the keys of _antennaViolations. The only change that helps here is checking if the net is special in inDbPostMoveInst callback.

eder-matheus commented:
> Thanks @eder-matheus that solves issue [OR-1027](https://openroad.atlassian.net/browse/OR-1027). However, I don't know why you need the extra nets_with_violations_ vector. It only holds the keys of _antennaViolations. The only change that helps here is checking if the net is special in inDbPostMoveInst callback.

I've changed it trying to clarify that portion of code, but it is consuming unnecessary memory. I've just reverted it.

maliberty commented on src/grt/src/AntennaRepair.cpp:
```diff
@@ -326,7 +326,7 @@ AntennaCbk::AntennaCbk(GlobalRouter* grouter) : _grouter(grouter)
 {
   for (odb::dbITerm* iterm : inst->getITerms()) {
-    if (iterm->getNet() != nullptr)
+    if (iterm->getNet() != nullptr && !iterm->getNet()->isSpecial())
       _grouter->addDirtyNet(iterm->getNet());
```
You could factor this to just call iterm->getNet() once.

maliberty commented:
@osamahammad21 I think you'll need to resolve your "requested changes" so Eder can merge.

osamahammad21 commented:
> @osamahammad21 I think you'll need to resolve your "requested changes" so Eder can merge.

Not really. It just needs to go through the CI
